cmd/paramgen/internal/paramgen_test.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/matryer/is"
+)
+
+func TestParseSpecificationSuccess(t *testing.T) {
+	testCases := []struct {
+		path string
+		name string
+		pkg  string
+		want map[string]sdk.Parameter
+	}{{
+		path: "./testdata/basic",
+		name: "SourceConfig",
+		pkg:  "example",
+		want: map[string]sdk.Parameter{
+			"globalConfig.foo": {
+				Default:     "bar",
+				Description: "foo is a required field in the global config with the name \"foo\" and default value \"bar\".",
+				Type:        sdk.ParameterTypeString,
+				Validations: []sdk.Validation{
+					sdk.ValidationRequired{},
+				},
+			},
+			"myString": {
+				Description: "myString my string description",
+				Type:        sdk.ParameterTypeString,
+			},
+			"myBool": {Type: sdk.ParameterTypeBool},
+			"myInt": {
+				Type: sdk.ParameterTypeInt,
+				Validations: []sdk.Validation{
+					sdk.ValidationLessThan{
+						Value: 100,
+					},
+					sdk.ValidationGreaterThan{
+						Value: 0,
+					},
+				},
+			},
+			"myUint":       {Type: sdk.ParameterTypeInt},
+			"myInt8":       {Type: sdk.ParameterTypeInt},
+			"myUint8":      {Type: sdk.ParameterTypeInt},
+			"myInt16":      {Type: sdk.ParameterTypeInt},
+			"myUint16":     {Type: sdk.ParameterTypeInt},
+			"myInt32":      {Type: sdk.ParameterTypeInt},
+			"myUint32":     {Type: sdk.ParameterTypeInt},
+			"myInt64":      {Type: sdk.ParameterTypeInt},
+			"myUint64":     {Type: sdk.ParameterTypeInt},
+			"myByte":       {Type: sdk.ParameterTypeString},
+			"myRune":       {Type: sdk.ParameterTypeInt},
+			"myFloat32":    {Type: sdk.ParameterTypeFloat},
+			"myFloat64":    {Type: sdk.ParameterTypeFloat},
+			"myDuration":   {Type: sdk.ParameterTypeDuration},
+			"myIntSlice":   {Type: sdk.ParameterTypeString},
+			"myFloatSlice": {Type: sdk.ParameterTypeString},
+			"myDurSlice":   {Type: sdk.ParameterTypeString},
+		},
+	},
+		{
+			path: "./testdata/complex",
+			name: "SourceConfig",
+			pkg:  "example",
+			want: map[string]sdk.Parameter{
+				"global.duration": {
+					Default:     "1s",
+					Description: "duration does not have a name so the type name is used.",
+					Type:        sdk.ParameterTypeDuration,
+				},
+				"nestMeHere.anotherNested": {
+					Type:        sdk.ParameterTypeInt,
+					Description: "nestMeHere.anotherNested is also nested under nestMeHere.",
+				},
+				"nestMeHere.formatThisName": {
+					Type:        sdk.ParameterTypeFloat,
+					Default:     "this is not a float",
+					Description: "formatThisName should become \"formatThisName\". Default is not a float but that's not a problem, paramgen does not validate correctness.",
+				},
+			},
+		},
+		{
+			path: "./testdata/tags",
+			name: "Config",
+			pkg:  "tags",
+			want: map[string]sdk.Parameter{
+				"innerConfig.my-name": {
+					Type:        sdk.ParameterTypeString,
+					Validations: []sdk.Validation{sdk.ValidationRequired{}},
+				},
+				"my-param": {
+					Type:        sdk.ParameterTypeInt,
+					Description: "my-param i am a parameter comment",
```

Comment:
Tiny nitpick: maybe have a test with a line comment as well?

Reply:
Good point 👍

Reply:
My comment wasn't really clear, I meant comments which come in the same line as the code, e.g.
```go
myField int // here's a comment
```
If the above makes any difference that is.
But good that you mentioned block comments as well. :+1:

Reply:
I added support for line comments. They are only used if the doc comment on the field does not exist. So like this:
```
type Config struct {
	// Foo has a doc comment
	Foo int // line comment is ignored
	Bar // Bar only has a line comment so it's used
}
```